src/Adapter/ImageType/ImageSettingsHelper.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\ImageType;
+
+use Image;
+use PrestaShop\PrestaShop\Core\ImageType\ImageSettingsHelperInterface;
+use Symfony\Component\Translation\TranslatorInterface;
+
+/**
+ * Helper functions for Image Settings page.
+ */
+class ImageSettingsHelper implements ImageSettingsHelperInterface
+{
+    private const MAX_EXECUTION_TIME_DEFAULT = 7200;
+
+    /**
+     * @var int
+     */
+    protected $max_execution_time = self::MAX_EXECUTION_TIME_DEFAULT;
+
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * @var string
+     */
+    public $productImagesDir;
+
+    /**
+     * @var bool
+     */
+    public $showDuplicatesAlert;
+
+    /**
+     * @param TranslatorInterface $translator
+     * @param string $productImagesDir
+     */
+    public function __construct(TranslatorInterface $translator, string $productImagesDir)
+    {
+        $this->translator = $translator;
+        $this->productImagesDir = $productImagesDir;
+        $this->showDuplicatesAlert = is_dir($productImagesDir . 'duplicates' . DIRECTORY_SEPARATOR);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function moveImagesToNewFileSystem(): array
```

Comment:
@ks129 I don't think we need it in 1.7, it's not possible to update directly from such an old version, right?

Reply:
I'm not sure. Maybe it's safer to remove it in 1.8?

Reply:
@ks129 This won't be needed, I am removing the functionality completely in https://github.com/PrestaShop/PrestaShop/pull/31311 👍

Reply:
@ks129 Merged, you can remove this. :-)